errors/handler.go

```diff
@@ -27,23 +31,26 @@ type StackFrames struct {
 	Package    string
 }
 
-var MarkSpanAsError = errors.New("")
+var markSpanAsError = errors.New("")
 
 // Write exception event in span using the recover data from panic
-func LogError(span opentracing.Span, recoverData interface{}, skipFrames int) {
-	var exceptionFields = getExceptionLogFields(recoverData, skipFrames+1)
-	span.LogFields(exceptionFields...)
+func WriteExceptionEvent(span opentracing.Span, recoverData interface{}, skipFrames int) {
 	span.SetTag("error", true)
+	if recoverData == markSpanAsError {
+		return
+	}
+	var exceptionFields = getExceptionLogFields("error", recoverData, skipFrames+1)
+	span.LogFields(exceptionFields...)
 }
 
-func LogErrorInRawSpan(rawSpan *tracer.RawSpan, err **errors.Error) {
+func WriteExceptionEventInRawSpan(rawSpan *tracer.RawSpan, err **errors.Error) {
 	if rawSpan.Tags == nil {
 		rawSpan.Tags = opentracing.Tags{}
 	}
-	if *err == MarkSpanAsError {
+	if *err == markSpanAsError {
 		rawSpan.Tags["error"] = true
```

Comment:
`rawSpan.Tags["error"] = true` is present in if clause and else clause.
Can we extract this outside of the if?

Reply:
You're right , I made the changes...